pkg/network/tracer/connection/kprobe/tcp_failed_conn_consumer.go

```diff
@@ -0,0 +1,103 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux_bpf
+// +build linux_bpf
+
+package kprobe
+
+import (
+	"sync"
+	"unsafe"
+
+	ddebpf "github.com/DataDog/datadog-agent/pkg/ebpf"
+	netebpf "github.com/DataDog/datadog-agent/pkg/network/ebpf"
+)
+
+type failedConnStats struct {
+	direction    netebpf.ConnDirection
+	failureCount uint64
+	lastErrno    int32
+}
+
+type failedConnMap map[netebpf.ConnTuple]failedConnStats
+
+type tcpFailedConnConsumer struct {
+	perfHandler *ddebpf.PerfHandler
+	failedConns failedConnMap
+
+	mutex             sync.Mutex
+	isRuntimeCompiled bool
+}
+
+func newTCPFailedConnConsumer(perfHandler *ddebpf.PerfHandler, isRuntimeCompiled bool) (*tcpFailedConnConsumer, error) {
+	return &tcpFailedConnConsumer{
+		perfHandler:       perfHandler,
+		failedConns:       make(failedConnMap),
+		isRuntimeCompiled: isRuntimeCompiled,
+	}, nil
+}
+
+func (c *tcpFailedConnConsumer) Start() {
+	if c == nil {
+		return
+	}
+
+	go func() {
+		for {
+			select {
+			case rawFailedConnStats, ok := <-c.perfHandler.DataChannel:
+				if !ok {
+					return
+				}
+
+				stats := toFailedConnStats(rawFailedConnStats.Data)
+				c.addFailedConn(stats)
+			}
+		}
+	}()
+}
+
+func (c *tcpFailedConnConsumer) Stop() {
+	if c == nil {
+		return
+	}
+
+	c.perfHandler.Stop()
+}
+
+// Returns the latest conn data
+func (c *tcpFailedConnConsumer) GetStats() failedConnMap {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+
+	res := c.failedConns
+	c.failedConns = make(failedConnMap)
+
+	return res
+}
+
+// Utils
+
+func (c *tcpFailedConnConsumer) addFailedConn(s *netebpf.FailedConnStats) {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+
+	stats := c.failedConns[(*s).Ct]
+	stats.failureCount += 1
+	stats.direction = netebpf.ConnDirection((*s).Dir)
+
+	if c.isRuntimeCompiled {
+		stats.lastErrno = (*s).Errno
```

Comment:
There is code in prebuilt that is sending this over? https://github.com/DataDog/datadog-agent/pull/13516/files#diff-abcc9d0eb0ef600b5f0f09242cda2c4744f11aad8c2fa2698b5bc64d570f61ecR172

Reply:
Good catch. I'll remove the errno reporting from the prebuilt version.

To give some context, this condition to make it runtime-compile is a very recent change I made. The CI was failing before due to the lack of offset guessing (everything looked good in the failing connection report except for the lack of the errno code). I chose to add this check rather than strip the whole feature from the prebuilt version.